gcc/testsuite/gcc.target/riscv/cv-mem-sw-compile-4.c

```diff
@@ -0,0 +1,33 @@
+/* { dg-do compile } */
+/* { dg-options "-march=rv32i_xcvmem -mabi=ilp32 -fno-unroll-loops" } */
+/* { dg-skip-if "" { *-*-* }  { "-O0" } { "" } } */
+
+/*
+ * Test for illegal normal register-immediate saves.
+ */
+
+int fooSIsigned (signed int* array_int, int n)
+{
+  int int_sum = 1;
+
+  for(int i=0; i<n; i++)
+  {
+    array_int[i] = int_sum;
+  }
+
+  return int_sum;
+}
+
+int fooSIunsigned (unsigned int* array_uint, int n)
+{
+  int uns_int_sum = 1;
+
+  for(int i=0; i<n; i++)
+  {
+    array_uint[i] = uns_int_sum;
+  }
+
+  return uns_int_sum;
+}
+
+/* { dg-final { scan-assembler-not "cv\\.sw\[\t \]+.*,\[0-9\]+\\(\[^!\]+\\)" } } */
```

Comment:
`cv-mem-lw-compile-4.c` is quite simple in comparison. I wonder if something similar would be suitable for `cv-mem-sw-compile-4.c` too such as:
```
void fooSIsigned (signed int* array_int, int i)
{
  array_int [3] = i;
}
```